### ISSUE TITLE
Error  Permission denied: '/tmp/filer/System Volume Information' (SMB Backup)

### ISSUE THREAD

**Issue by de1m** [borgbackup/borg#3337]
Hi,

If I run a backupjob to save my fileshare (mouted SMB in ubuntu), I get the error above.

My command for backup:
````
BORG_PASSPHRASE=PASS BORG_CACHE_DIR=/backup/borg borg create \
    -C zlib --json  --pattern "+ /tmp/filer/usershare" --pattern "+ /tmp/filer/publicshare" \
    --pattern "+ /tmp/filer/Geschäftsleitung" --pattern "+ /tmp/filer/backup"  \
    --pattern "- /tmp/filer/publicshare/Softwarepool" --pattern "- /tmp/filer/*" \
    --pattern "- /tmp/filer/System Volume Information" \
    /backup/filer::15Nov2017_591 /tmp/filer

Connection through backup2.aditosoftware.local to backup2.aditosoftware.local 
ERR: /tmp/filer/System Volume Information: scandir: [Errno 13] Permission denied: '/tmp/filer/System Volume Information'
```` 

The pattern "--pattern "- /tmp/filer/System Volume Information" not working?

P.S.: I used the borg version 1.1.1 on ubuntu 16.04.

**Comment by ThomasWaldmann** [comment]
> --pattern "- /tmp/filer/*"
> --pattern "- /tmp/filer/System Volume Information"

Looks like the first already matches everything the second would match.

Check the docs again about how exactly matching is done, order and match types matter here.

>ERR: /tmp/filer/System Volume Information: > scandir: [Errno 13] Permission> denied: '/tmp/filer/System Volume Information' |
> 
> The patter "--pattern "- /tmp/filer/System Volume Information" not working?

From 1.1.2 changelog:

- create: don't do stat() call on excluded-norecurse directory, fix exception handling for stat() call, #3209

- mention "!" (exclude-norecurse) type in "patterns" help

You will need that "!" type of pattern for the "Permission denied" folder (not "-").

**Comment by Siassei** [comment]
holy moly, hard to read a single shell line + I thing the variables should not working without export
```bash
export BORG_PASSPHRASE=PASS
export BORG_CACHE_DIR=/backup/borg

borg create                                              \
  -C zlib                                                \
  --json                                                 \
  --pattern "+ /tmp/filer/usershare"                     \
  --pattern "+ /tmp/filer/publicshare"                   \
  --pattern "+ /tmp/filer/Geschäftsleitung"              \
  --pattern "+ /tmp/filer/backup"                        \
  --pattern "- /tmp/filer/publicshare/Softwarepool"      \
  --pattern "- /tmp/filer/*"                             \
  --pattern "- /tmp/filer/System Volume Information"     \                                               
                                                         \
  /backup/filer::15Nov2017_591 

/tmp/filer Connection through backup2.aditosoftware.local to backup2.aditosoftware.local 
ERR: /tmp/filer/System Volume Information: scandir: [Errno 13]
Permission denied: '/tmp/filer/System Volume Information'
```

Please add the lines to get more output information
`--verbose --list --show-rc`

I had test it with a example project as bellow listed. There is not a problem with borg or borg-pattern-match! It is a permission problem on your local machine. Please check all premissions (your user, borg user, ...) and check if there are no miss writings in your code.

```bash
$ borg --version
borg 1.1.2

$ mkdir /tmp/a
$ mkdir /tmp/a/b
$ mkdir "/tmp/a/sdf df"
$ touch "/tmp/a/sdf df/df"
$ touch /tmp/a/sdgf
$ touch /tmp/a/b/dfghr

$ borg init --encryption=repokey /tmp/borg-1
$ borg create                       \
    --verbose                       \
    --list                          \
    --show-rc                       \
    --stats                         \
    -C zlib                         \
    --json                          \
    --pattern "- /tmp/a/sdf df/*"   \
    --pattern "- /tmp/a/sdf df/df"  \
    --stats                         \
                                    \
    /tmp/borg-1::Test5              \
    /tmp/a

Enter passphrase for key /tmp/borg-1: 
A /tmp/a/b/dfghr
d /tmp/a/b
x /tmp/a/sdf df/df
d /tmp/a/sdf df
U /tmp/a/sdgf
d /tmp/a
{
...
}
terminating with success status, rc 0
```

**Comment by ThomasWaldmann** [comment]
@Siassei shell vars do work without export when mentioned on same line, directly before the command.

And the permissions issue will be fixed when using borg 1.1.2 and using --pattern "! /folder/without/permission".

**Comment by de1m** [comment]
@ThomasWaldmann thank you for the info. I've updated my borg version to 1.1.2 and change the command (added !)
It worked with the command:
````
sudo BORG_PASSPHRASE=PASS BORG_CACHE_DIR=/backup/borg borg create \
-C zlib --json \
 --pattern "+ /tmp/filer/usershare" \
--pattern "+ /tmp/filer/publicshare" \
--pattern "+ /tmp/filer/Geschäftsleitung" \
--pattern "+ /tmp/filer/backup"  \
--pattern "! /tmp/filer/*" \
/backup/filer::20Nov2017_125 /tmp/filer 
````
But I've a couple of files, with another permissions, and the backup user don't have permissions for this. So the backup jobs will'be stopped on this files, it's so the same error.
My question ist now, can I say borg to ignore the error?

**Comment by ThomasWaldmann** [comment]
@de1m when borg tries to read input files and there is a problem, it usually logs the problem and skips the file.

So, please double check if it really "stops" / "aborts" your backup or if it just logs/skips.

Using --list might be useful for that.

**Comment by de1m** [comment]
okay, I've checked - the files from yesterday exists.

My problem is that I use the borg in a nodejs program [Nodebackup](https://github.com/aditosoftware/nodebackup) and this exited with a error, can I say borg to ignore the error (exit code 0 and with --json output) ?

**Comment by ThomasWaldmann** [comment]
borg will exit with rc 1 if it has to skip files (due to permission or other problems that make them unreadable).

rc 1 means "warning" (not: "error", which would be rc 2) and that's what it should be.
warning often means "it was somehow okish as borg reached the end of the recursion (so it did not e.g. crash suddenly), but there were some problems".

There are 2 ways you can get rc 0:
- either get higher permissions so you do not run into permission errors or
-  exclude the files / dirs you do not have permission to read/access.

I'm closing this issue as the fundamental problem was fixed with borg 1.1.2 and using "!".
